lib/ansible/module_utils/network/nxos/config/bfd_interfaces/bfd_interfaces.py

```diff
@@ -59,18 +59,35 @@ def execute_module(self):
         warnings = list()
         cmds = list()
 
-        existing_bfd_interfaces_facts, platform = self.get_bfd_interfaces_facts()
-        cmds.extend(self.set_config(existing_bfd_interfaces_facts, platform))
-        if cmds:
+        if self.state in self.ACTION_STATES:
+            existing_bfd_interfaces_facts, platform = self.get_bfd_interfaces_facts()
+        else:
+            existing_interfaces_facts, platform = [], ''
+
+        if self.state in self.ACTION_STATES or self.state == 'rendered':
+            cmds.extend(self.set_config(existing_bfd_interfaces_facts, platform))
```

Comment:
@ganeshrn gave a run for `rendered` state, and it's throwing `local variable referenced before assignment` error.